benchmarks/serde-benchmarks/build.gradle.kts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+import software.amazon.smithy.gradle.tasks.SmithyBuild
+
+plugins {
+    kotlin("multiplatform")
+    id("org.jetbrains.kotlinx.benchmark")
+    id("software.amazon.smithy")
+}
+
+buildscript {
+    val smithyVersion: String by project
+    dependencies {
+        classpath("software.amazon.smithy:smithy-cli:$smithyVersion")
+    }
+}
+
+
+extra.set("skipPublish", true)
+
+
+val platforms = listOf("common", "jvm")
+
+platforms.forEach { platform ->
+    apply(from = rootProject.file("gradle/${platform}.gradle"))
+}
+
+val experimentalAnnotations = listOf("kotlin.RequiresOptIn", "aws.smithy.kotlin.runtime.util.InternalApi")
+
+kotlin {
+    sourceSets {
+        all {
+            val srcDir = if (name.endsWith("Main")) "src" else "test"
+            val resourcesPrefix = if (name.endsWith("Test")) "test-" else  ""
+            // the name is always the platform followed by a suffix of either "Main" or "Test" (e.g. jvmMain, commonTest, etc)
+            val platform = name.substring(0, name.length - 4)
+            kotlin.srcDir("$platform/$srcDir")
+            resources.srcDir("$platform/${resourcesPrefix}resources")
+            languageSettings.progressiveMode = true
+            experimentalAnnotations.forEach { languageSettings.optIn(it) }
+        }
+
+        val kotlinxBenchmarkVersion: String by project
+        val coroutinesVersion: String by project
+        commonMain {
+            dependencies {
+                implementation("org.jetbrains.kotlinx:kotlinx-benchmark-runtime:$kotlinxBenchmarkVersion")
+                implementation(project(":runtime:serde:serde-json"))
+                implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutinesVersion")
+            }
+        }
+        val jvmMain by getting {
+            kotlin.srcDir("${project.buildDir}/generated-src/src")
+        }
+    }
+}
+
+benchmark {
+    targets {
+        register("jvm")
+    }
+
+    configurations {
```

Comment:
question

Is the idea here that we may define multiple benchmarking configurations, like say one perhaps specifically for CI?  How are the configurations specified at execution time? (I searched but was unable to come up w/ suitably specific terms)

Reply:
Had to look it up myself, closest documentation I could find on it:
```
    configurations {
        main { 
            // configure default configuration
        }
        smoke { 
            // create and configure "smoke" configuration, e.g. with several fast benchmarks to quickly check
            // if code changes result in something very wrong, or very right. 
        }       
    }
```

So yeah it looks like if you wanted to multiple benchmark classifications you could group them into a single config and run them independent of the other.